src/UI/Component/Layout/Factory.php

```diff
@@ -0,0 +1,182 @@
+     *     or by clicking a different button.
+     *
+     *   rivals:
+     *     Tab Bar: >
+     *       The sidebar (and its components) shall not be used to substitute
+     *       functionality available at objects, such as settings, members or
+     *       learning progress. Those remain in the Tab Bar.
+     *
+     *     Content Actions: >
+     *       Also, adding new items, the actions-menu (with comments, notes and tags),
+     *       moving, linking or deleting objects and the like are not part of
+     *       the sidebar.
+     *
+     *     Personal Desktop: >
+     *       The Personal Desktop provides access to services and tools and
+     *       displays further information at first glance (e.g. the calendar).
+     *       The sidebar may reference those tools as well, but rather in form
+     *       of a link than a widget.
+     *
+     *     Notification Center: >
+     *       Notifications of the system to the user, e.g. new Mail, are placed
+     *       in the Notification Center.
+     *       The direction of communication for the sidebar is "user to system",
+     *       while the direction is "system to user" in the Notification Center.
+     *       However, navigation from both components can lead to the same page.
+     *
+     *     Modal: >
+     *       Forms with the intention of modifying the content are placed in modals
+     *       or on the content-page.
+     *
+     * rules:
+     *   usage:
+     *     1: There SHOULD be a sidebar on the page. Kiosk-Mode is the only exception.
+     *     2: If there is a sidebar, it MUST be unique for the page.
+     *
+     *   composition:
+     *     1: The bar MUST NOT contain items other than iconographic buttons.
+     *     2: The bar MUST contain at least one iconographic button.
+     *     3: The bar SHOULD NOT contain more than five iconographic buttons.
+     *
+     *   style:
+     *     1: The bar MUST have a fixed witdth (desktop).
+     *
+     *   interaction:
+     *     1: >
+     *        Operating elements in the bar MUST either lead to further
+     *        navigational options within the bar (open a slate)
+     *        OR actually invoke navigation, i.e. change the location/content
+     *        of the current page.
+     *     2: Elements in the bar MUST NOT open a modal or window.
+     *
+     * ----
+     *
+     * @param  \ILIAS\UI\Component\Layout\SidebarEntry[]    $entries
+     * @param  int|null  $active
+     * @return  \ILIAS\UI\Component\Layout\Sidebar
+     */
+    public function sidebar(array $entries, $active=null);
+
+
+        /**
+     * ---
+     * description:
+     *   purpose: >
+     *     The sidebar entry bundles an iconographic button and a slate.
+     *   composition: >
+     *     There is no composition (=visual appearance) of this component itself -
+     *     its button will be rendered in the sidebar with a signal triggering
+     *     its slate.
+     *
+     * ----
```

Comment:
Ich denke hier sollten wir dokumentieren, dass der `sidebarEntry` den Button mit der Slate verknüpft, oder diese Verknüpfung nicht in dem `SidebarEntry` sondern erst kurz vor knapp machen. Letztlich ist es irritierend, das dies nicht funktioniert:

```
$entry = $f->sidebarEntry($button);
$button2 = $entry->getButton();
assert($button === $button2);
```